src/transformers/models/auto/feature_extraction_auto.py

```diff
@@ -142,22 +142,28 @@ def from_pretrained(cls, pretrained_model_name_or_path, **kwargs):
             os.path.join(pretrained_model_name_or_path, FEATURE_EXTRACTOR_NAME)
         )
 
-        if not is_feature_extraction_file and not is_directory:
+        has_local_config = (
+            os.path.exists(os.path.join(pretrained_model_name_or_path, CONFIG_NAME)) if is_directory else False
+        )
+
+        # load config, if it can be loaded
+        if not is_feature_extraction_file and (has_local_config or not is_directory):
             if not isinstance(config, PretrainedConfig):
                 config = AutoConfig.from_pretrained(pretrained_model_name_or_path, **kwargs)
 
         kwargs["_from_auto"] = True
         config_dict, _ = FeatureExtractionMixin.get_feature_extractor_dict(pretrained_model_name_or_path, **kwargs)
 
         model_type = config_class_to_model_type(type(config).__name__)
+
         if model_type is not None:
             return FEATURE_EXTRACTOR_MAPPING[type(config)].from_dict(config_dict, **kwargs)
         elif "feature_extractor_type" in config_dict:
             feature_extractor_class = feature_extractor_class_from_name(config_dict["feature_extractor_type"])
             return feature_extractor_class.from_dict(config_dict, **kwargs)
 
         raise ValueError(
-            f"Unrecognized model in {pretrained_model_name_or_path}. Should have a `feature_extractor_type` key in "
```

Comment:
The error message seemed wrong to me...cc @sgugger . Corrected it here. I think this was copy pasted from `AutoConfig` which does "name" fallback pattern matching, but for feature extractors I think we go over the `model_type` of the `config.json`

Reply:
Thanks for fixing!